pairing/bn256/constants.go

```diff
@@ -0,0 +1,56 @@
+package bn256
+
+import (
+	"math/big"
+)
+
+func bigFromBase10(s string) *big.Int {
+	n, _ := new(big.Int).SetString(s, 10)
+	return n
+}
+
+// u is the BN parameter that determines the prime: 1868033³.
```

Comment:
Hmm - this is not a prime - so why does it `determine the prime`?

Reply:
There are polynomials that give the trace of the frobenius, the order of the curve, and the characteristic of the field in terms of this parameter. This is how BN curves are constructed.